src/CameraServer.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of the Camera Streaming Daemon
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+#include <cstring>
+#include <set>
+#include <sstream>
+#include <string.h>
+
+#include "CameraComponent_V4L2.h"
+#include "CameraServer.h"
+#include "log.h"
+#include "mavlink_server.h"
+#include "util.h"
+
+#define DEVICE_PATH "/dev/"
+#define VIDEO_PREFIX "video"
+#define DEFAULT_SERVICE_PORT 8554
+
+CameraServer::CameraServer(ConfFile &conf)
+    : mavlink_server(conf, streams, rtsp_server)
+    , rtsp_server(streams, DEFAULT_SERVICE_PORT)
+    , cameraCount(0)
+{
+    cameraCount = detectCamera(conf);
```

Comment:
I just don't get why to limit this functionality to V4L2 cameras and why not using the camera discovery mechanism that is used to create Streams.
Won't CameraServer support Realsense RGB, IR and depth streams?